File: CS/S132265/MyLayoutControlItemPropertyWrapper.cs
using DevExpress.XtraLayout;
using DevExpress.XtraEditors.Mask;
using System.ComponentModel;
using DevExpress.XtraEditors;

namespace DXSample {
    public class MyLayoutControlItemPropertyWrapper : LayoutControlItemWrapper {
        protected TextEdit Editor { get { return Item.Control as TextEdit; } }

        [Category("Mask")]
        [DefaultValue(AutoCompleteType.Default)]
        public AutoCompleteType AutoComplete {
            get {
                if (Editor != null) return Editor.Properties.Mask.AutoComplete;
                else return AutoCompleteType.Default;
            }
            set { if (Editor != null) Editor.Properties.Mask.AutoComplete = value; }
        }

        [Category("Mask")]
        [DefaultValue(false)]
        public bool BeepOnError {
            get {
                if (Editor != null) return Editor.Properties.Mask.BeepOnError;
                else return false;
            }
            set { if (Editor != null) Editor.Properties.Mask.BeepOnError = value; }
        }

        [Category("Mask")]
        [DefaultValue("")]
        public string EditMask {
            get {
                if (Editor != null) return Editor.Properties.Mask.EditMask;
                else return string.Empty;
            }
            set { if (Editor != null) Editor.Properties.Mask.EditMask = value; }
        }

        [Category("Mask")]
        [DefaultValue(true)]
        public bool IgnoreMaskBlank {
            get {
                if (Editor != null) return Editor.Properties.Mask.IgnoreMaskBlank;
                else return true;
            }
            set { if (Editor != null) Editor.Properties.Mask.IgnoreMaskBlank = true; }
        }

        [Category("Mask")]
        [DefaultValue(MaskType.None)]
        public MaskType MaskType {
            get {
                if (Editor != null) return Editor.Properties.Mask.MaskType;
                else return MaskType.None;
            }
            set { if (Editor != null) Editor.Properties.Mask.MaskType = value; }
        }

        [Category("Mask")]
        [DefaultValue('\0')]
        public char PlaceHolder {
            get {
                if (Editor != null) return Editor.Properties.Mask.PlaceHolder;
                else return '\0';
            }
            set { if (Editor != null) Editor.Properties.Mask.PlaceHolder = '\0';  }
        }

        [Category("Mask")]
        [DefaultValue(true)]
        public bool SaveLiteral {
            get {
                if (Editor != null) return Editor.Properties.Mask.SaveLiteral;
                else return true;
            }
            set { if (Editor != null) Editor.Properties.Mask.SaveLiteral = value; }
        }

        [Category("Mask")]
        [DefaultValue(true)]
        public bool ShowPlaceHolders {
            get {
                if (Editor != null) return Editor.Properties.Mask.ShowPlaceHolders;
                else return true;
            }
            set { if (Editor != null) Editor.Properties.Mask.ShowPlaceHolders = value; }
        }

        [Category("Mask")]
        [DefaultValue(false)]
        public bool UseMaskAsDisplayFormat {
            get {
                if (Editor != null) return Editor.Properties.Mask.UseMaskAsDisplayFormat;
                else return false;
            }
            set { if (Editor != null) Editor.Properties.Mask.UseMaskAsDisplayFormat = value; }
        }
    }
}

File: VB/S132265/MyLayoutControlItemPropertyWrapper.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraLayout
Imports DevExpress.XtraEditors.Mask
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace DXSample
	Public Class MyLayoutControlItemPropertyWrapper
		Inherits LayoutControlItemWrapper
		Protected ReadOnly Property Editor() As TextEdit
			Get
				Return TryCast(Item.Control, TextEdit)
			End Get
		End Property

		<Category("Mask"), DefaultValue(AutoCompleteType.Default)> _
		Public Property AutoComplete() As AutoCompleteType
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.AutoComplete
				Else
					Return AutoCompleteType.Default
				End If
			End Get
			Set(ByVal value As AutoCompleteType)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.AutoComplete = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(False)> _
		Public Property BeepOnError() As Boolean
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.BeepOnError
				Else
					Return False
				End If
			End Get
			Set(ByVal value As Boolean)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.BeepOnError = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue("")> _
		Public Property EditMask() As String
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.EditMask
				Else
					Return String.Empty
				End If
			End Get
			Set(ByVal value As String)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.EditMask = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(True)> _
		Public Property IgnoreMaskBlank() As Boolean
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.IgnoreMaskBlank
				Else
					Return True
				End If
			End Get
			Set(ByVal value As Boolean)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.IgnoreMaskBlank = True
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(MaskType.None)> _
		Public Property MaskType() As MaskType
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.MaskType
				Else
					Return MaskType.None
				End If
			End Get
			Set(ByVal value As MaskType)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.MaskType = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(ControlChars.NullChar)> _
		Public Property PlaceHolder() As Char
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.PlaceHolder
				Else
					Return ControlChars.NullChar
				End If
			End Get
			Set(ByVal value As Char)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.PlaceHolder = ControlChars.NullChar
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(True)> _
		Public Property SaveLiteral() As Boolean
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.SaveLiteral
				Else
					Return True
				End If
			End Get
			Set(ByVal value As Boolean)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.SaveLiteral = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(True)> _
		Public Property ShowPlaceHolders() As Boolean
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.ShowPlaceHolders
				Else
					Return True
				End If
			End Get
			Set(ByVal value As Boolean)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.ShowPlaceHolders = value
				End If
			End Set
		End Property

		<Category("Mask"), DefaultValue(False)> _
		Public Property UseMaskAsDisplayFormat() As Boolean
			Get
				If Editor IsNot Nothing Then
					Return Editor.Properties.Mask.UseMaskAsDisplayFormat
				Else
					Return False
				End If
			End Get
			Set(ByVal value As Boolean)
				If Editor IsNot Nothing Then
					Editor.Properties.Mask.UseMaskAsDisplayFormat = value
				End If
			End Set
		End Property

		Public Overrides Function Clone() As BasePropertyGridObjectWrapper
			Return New MyLayoutControlItemPropertyWrapper()
		End Function
	End Class
End Namespace